algoliasearch/src/main/java/com/algolia/search/clients/AnalyticsClient.java

```diff
@@ -0,0 +1,314 @@
+  }
+
+  /**
+   * Get an A/B test information and results.
+   *
+   * @param id The ABTest ID
+   * @param requestOptions Options to pass to this request
+   */
+  public ABTest getABTest(long id, RequestOptions requestOptions) {
+    return LaunderThrowable.unwrap(getABTestAsync(id, requestOptions));
+  }
+
+  /**
+   * Get an A/B test information and results.
+   *
+   * @param id The ABTest ID
+   */
+  public CompletableFuture<ABTest> getABTestAsync(long id) {
+    return getABTestAsync(id, null);
+  }
+
+  /**
+   * Get an A/B test information and results.
+   *
+   * @param id The ABTest ID
+   * @param requestOptions Options to pass to this request
+   */
+  public CompletableFuture<ABTest> getABTestAsync(long id, RequestOptions requestOptions) {
+    return transport.executeRequestAsync(
+        HttpMethod.GET, "/2/abtests/" + id, CallType.READ, null, ABTest.class, requestOptions);
+  }
+
+  /**
+   * Marks the A/B Test as stopped. At this point, the test is over and cannot be restarted. As a
+   * result, your application is back to normal: index A will perform as usual, receiving 100% of
+   * all search requests. Associated metadata and metrics are still stored
+   *
+   * @param id The ABTest ID
+   */
+  public StopAbTestResponse stopABTest(long id) {
+    return LaunderThrowable.unwrap(stopABTestAsync(id, null));
+  }
+
+  /**
+   * Marks the A/B Test as stopped. At this point, the test is over and cannot be restarted. As a
+   * result, your application is back to normal: index A will perform as usual, receiving 100% of
+   * all search requests. Associated metadata and metrics are still stored
+   *
+   * @param id The ABTest ID
+   * @param requestOptions Options to pass to this request
+   */
+  public StopAbTestResponse stopABTest(long id, RequestOptions requestOptions) {
+    return LaunderThrowable.unwrap(stopABTestAsync(id, requestOptions));
+  }
+
+  /**
+   * Marks the A/B Test as stopped. At this point, the test is over and cannot be restarted. As a
+   * result, your application is back to normal: index A will perform as usual, receiving 100% of
+   * all search requests. Associated metadata and metrics are still stored
+   *
+   * @param id The ABTest ID
+   */
+  public CompletableFuture<StopAbTestResponse> stopABTestAsync(long id) {
+    return stopABTestAsync(id, null);
+  }
+
+  /**
+   * Marks the A/B Test as stopped. At this point, the test is over and cannot be restarted. As a
+   * result, your application is back to normal: index A will perform as usual, receiving 100% of
+   * all search requests. Associated metadata and metrics are still stored
+   *
+   * @param id The ABTest ID
+   * @param requestOptions Options to pass to this request
+   */
+  public CompletableFuture<StopAbTestResponse> stopABTestAsync(
+      long id, RequestOptions requestOptions) {
+    return transport.executeRequestAsync(
+        HttpMethod.POST,
+        "/2/abtests/" + id + "/stop",
```

Comment:
I would make sure `id` is properly escaped with a URL encoding. Same comment for everytime we are building URLs

Reply:
I'd recommend something similar to what I did in PHP. I defined a helper method to escape all variables. It looks like a sprintf call.

```php
ap_path("/2/abtests/%s/stop", id)
api_path('/1/indexes/%s/facets/%s/query', indexName, facetName),
```

https://github.com/algolia/algoliasearch-client-php/blob/master/src/Support/Helpers.php#L26-L35

Reply:
Thanks 👍 I have did the same in .NET client. I will add it there.